zerver/lib/upload/s3.py

```diff
@@ -108,16 +109,28 @@ def upload_content_to_s3(
     )
 
 
+BOTO_CLIENT: S3Client | None = None
+
+
+def get_boto_client() -> S3Client:
+    """
+    Creating the client takes a long time so we need to cache it.
+    """
+    global BOTO_CLIENT
+    if BOTO_CLIENT is None:
+        BOTO_CLIENT = get_bucket(settings.S3_AUTH_UPLOADS_BUCKET).meta.client
```

Comment:
The way we get `boto_client` is different as compared to #21725 because of the changes we made in https://github.com/zulip/zulip/commit/3aea67a8ede6b8493f1faaa361556e376e94e203